x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/last_value.tsx

```diff
@@ -0,0 +1,217 @@
+  if (indexPattern.timeFieldName) {
+    dateFields.sort(({ name: nameA }, { name: nameB }) => {
+      if (nameA === indexPattern.timeFieldName) {
+        return -1;
+      }
+      if (nameB === indexPattern.timeFieldName) {
+        return 1;
+      }
+      return 0;
+    });
+  }
+  return dateFields;
+}
+
+export interface LastValueIndexPatternColumn extends FieldBasedIndexPatternColumn {
+  operationType: 'last_value';
+  params: {
+    sortField: string;
+    sortOrder: 'asc' | 'desc';
+    // last value on numeric fields can be formatted
+    format?: {
+      id: string;
+      params?: {
+        decimals: number;
+      };
+    };
+  };
+}
+
+export const lastValueOperation: OperationDefinition<LastValueIndexPatternColumn, 'field'> = {
+  type: 'last_value',
+  displayName: i18n.translate('xpack.lens.indexPattern.lastValue', {
+    defaultMessage: 'Last value',
+  }),
+  input: 'field',
+  onFieldChange: (oldColumn, field) => {
+    const newParams = { ...oldColumn.params };
+
+    if ('format' in newParams && field.type !== 'number') {
+      delete newParams.format;
+    }
+    return {
+      ...oldColumn,
+      dataType: field.type as DataType,
+      label: ofName(field.displayName),
+      sourceField: field.name,
+      params: newParams,
+    };
+  },
+  getPossibleOperationForField: ({ aggregationRestrictions, type }) => {
+    if (supportedTypes.has(type) && !aggregationRestrictions) {
+      return { dataType: type as DataType, isBucketed: false, scale: 'ordinal' };
+    }
+  },
+  getDisabledStatus(indexPattern: IndexPattern) {
+    const hasDateFields = indexPattern && getDateFields(indexPattern).length;
+    if (!hasDateFields) {
+      return i18n.translate('xpack.lens.indexPattern.lastValue.disabled', {
+        defaultMessage: 'This function requires the presence of a date field in your index',
+      });
+    }
+  },
+  buildColumn({ field, previousColumn, indexPattern }) {
+    const sortField =
+      indexPattern.timeFieldName ||
+      indexPattern.fields.find((f) => f.type === 'date')?.name ||
+      field.name;
+
+    const sortOrder =
+      (previousColumn?.params &&
+        'sortOrder' in previousColumn.params &&
+        previousColumn?.params?.sortOrder) ||
+      'desc';
+
+    return {
+      label: ofName(field.displayName),
+      dataType: field.type as DataType,
+      operationType: 'last_value',
+      isBucketed: false,
+      scale: 'ratio',
+      sourceField: field.name,
+      params: {
+        sortOrder,
+        sortField,
+      },
+    };
+  },
+  toEsAggsConfig: (column, columnId) => ({
+    id: columnId,
+    enabled: true,
+    schema: 'metric',
+    type: 'top_hits',
+    params: {
+      field: column.sourceField,
+      aggregate: 'concat',
+      size: 1,
+      sortOrder: column.params.sortOrder,
+      sortField: column.params.sortField,
+    },
+  }),
+
+  isTransferable: (column, newIndexPattern) => {
+    const newField = newIndexPattern.getFieldByName(column.sourceField);
```

Comment:
Please also check whether the other index pattern has `column.params.sourceField` as well, otherwise this will keep an invalid operation in some cases